pyveda/vedaset/store/vedabase.py

```diff
@@ -81,15 +81,15 @@ def batch_generator(self, batch_size, steps=None, loop=True, shuffle=True, chann
                                 image_transform=image_transform,
                                 pad=pad, **kwargs)
 
-    def clean(self, count=None, include_background_tiles=True):
+    def clean(self, fname, count=None, include_background_tiles=True):
```

Comment:
Can we update this docstring to clarify what it means when we call `delete` or whatever on the datasample? That is, `delete` _only_ writes to the remote dataset, and no modification whatsoever happens on the underlying vedabase file. This will help people who might be confused when they go to iterate over their vedabase after cleaning it and are surprised to see that nothing has changed, and they should expect that their local vedaset will be inconsistent with the vedaset on Veda because we don't implement the same changes locally.  It could be good also to suggest a reasonable path to mitigating this inconsistency, eg, once cleaning/deleting/editing is complete, they should just re-download their vedabase from the updated dataset on Veda, and then local/remote consistency will be restored

Reply:
I'm still confused, can you give me a quick Python example of this in action? So you're iterating over a local vedabase, and you call delete on the datapoint you're on somehow, and that thing doesn't delete, it's another thing somewhere else?

Reply:
@drwelby right, which is confusing behavior at the api level, which is why the docstrings here are very important. Effectively means there's no difference between cleaning on a basic VCP vs a VedaBase. In that respect we have somewhat confusing functionality in place but it means that the basic patterns for making api calls to veda from a vedabase are in place (as well as the patterns for correctly iterating over (image, label, id) samples from a local file which is good. I feel like this could be viewed as a progress point to adding the kind of functionality we'd like to see here, which handles local modifications in sync, but if we have some good docs in place about the current state of what the method does and doesn't do, we can merge in and benefit from a lot of the other solid work @chloeahampton done like adding dataset ids, vb iteration patterns, etc. That's my take from what I know from working with Chloe but she can elaborate

Reply:
If the behavior is confusing I'd rather see it named to something closer to what it actually does,  like `delete_from_server` or ???

Reply:
that sounds reasonable

Reply:
wait, change `clean` to `remove_from_server`? I don't like that, because you're not removing all of the datapoints from the server, just a select few. But I see what you're saying. I could change the `remove_dp` to `remove_from_server` then add some docstring clarification. Does that work?

Reply:
Yeah, not the whole `clean` method. Or I could see Vedabase rows having a `remote` attribute that stores a reference to the VCP datapoint so you could update/delete it like `row.remote.delete()`

Reply:
The remote attribute would be cool, but I don't think that work is in the scope of this PR--that would require quite a bit of wrangling, I think. For now, I can just change the name of the method here.

Reply:
I disagree, @drwelby 's suggestion ought to be incorporated into the scope of this work. It's not very much code to implement this, and it would solve the confusion around the api for this work